src/vm/luca_loaderwriter.cpp

```diff
@@ -0,0 +1,75 @@
+#include "common.h"
+#include "luca_loaderwriter.h"
+#include "pal.h"
+
+LoaderWriter::LoaderWriter(int pid) {
+    LIMITED_METHOD_CONTRACT;
+
+    this->pid = pid;
+
+    SString path;
+    path.Printf("loader-%d.map", pid);
+    this->OpenFile(path);
+}
+
+void LoaderWriter::WriteRecord(SString& command, SString& path, SString& guid) {
+
+    CONTRACTL{
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    SString line;
+    line.Printf("%S;%S;%S", command, path, guid);
+
+    EX_TRY
+    {
+        StackScratchBuffer scratch;
+        const char* strLine = line.GetANSI(scratch);
+        ULONG inCount = line.GetCount();
+        ULONG outCount;
+
+        this->stream->Write(strLine, inCount, &outCount);
```

Comment:
You should make sure that stream != NULL before you try to write to it, since it might be NULL if open failed.